### PR TITLE
Consistently use Kubebuilder tags in Eventing reconcilers

### PR DESCRIPTION
**Pull request by pxsalehi**
**Description**

Changes proposed in this pull request:

* Consistently use Kubebuilder tags in Eventing reconcilers.
* Remove unused cluster role rule for `natsclusters`.

### CODE REVIEW

File: components/eventing-controller/controllers/subscription/beb/reconciler.go [hunk 1]
```diff
@@ -92,10 +92,11 @@ func NewReconciler(ctx context.Context, client client.Client, logger *logger.Log
 
 // +kubebuilder:rbac:groups=eventing.kyma-project.io,resources=subscriptions,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=eventing.kyma-project.io,resources=subscriptions/status,verbs=get;update;patch
-// Generate required RBAC to emit kubernetes events in the controller
+// Generate required RBAC to emit kubernetes events in the controller.
 // +kubebuilder:rbac:groups="",resources=events,verbs=create;patch
 // +kubebuilder:rbac:groups=gateway.kyma-project.io,resources=apirules,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:printcolumn:name="Ready",type=bool,JSONPath=`.status.Ready`
```
Comment (nachtmaar) [inline components/eventing-controller/controllers/subscription/beb/reconciler.go:98]: Can you explain why this is not required anymore ? Does it mean the Ready column will get removed ?

**Review by nachtmaar — COMMENTED** [review]
LGTM. Just one question :)

File: components/eventing-controller/controllers/subscription/beb/reconciler.go (hunk 1 above)
Comment (pxsalehi) [reply components/eventing-controller/controllers/subscription/beb/reconciler.go:98]: https://book.kubebuilder.io/reference/markers.html

Comment (nachtmaar) [reply components/eventing-controller/controllers/subscription/beb/reconciler.go:98]: Makes sense! I see it on the go struct here: https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/api/v1alpha1/subscription_types.go#L223

**Review by nachtmaar — APPROVED** [review]

**Review by nachtmaar — DISMISSED** [review]

**Review by nachtmaar — APPROVED** [review]

**Pull request merged**